src/sql/base/browser/ui/taskbar/actionbar.ts

```diff
@@ -128,15 +143,87 @@ export class ActionBar extends ActionRunner implements IActionRunner {
 		this._actionsList = document.createElement('ul');
 		this._actionsList.className = 'actions-container';
 		this._actionsList.setAttribute('role', 'toolbar');
+		this._actionsList.id = 'actions-container';
 		if (this._options.ariaLabel) {
 			this._actionsList.setAttribute('aria-label', this._options.ariaLabel);
 		}
 
 		this._domNode.appendChild(this._actionsList);
 
+		if (this._collapseOverflow) {
+			this._overflow = document.createElement('ul');
+			this._overflow.id = 'overflow';
+			this._overflow.className = 'overflow';
+			this._domNode.appendChild(this._overflow);
+		} else {
+			this._actionsList.style.flexWrap = 'wrap';
+		}
+
 		container.appendChild(this._domNode);
 	}
 
+	@debounce(300)
+	private resizeToolbar() {
```

Comment:
can we somehow make this implementation generic and reusable by other toolbars?

Reply:
will do this in next PR